src/main/java/org/jabref/gui/JabRefDialogService.java

```diff
@@ -226,16 +224,7 @@ public void showErrorDialogAndWait(FetcherException fetcherException) {
         String localizedMessage = fetcherException.getLocalizedMessage();
         Optional<SimpleHttpResponse> httpResponse = fetcherException.getHttpResponse();
         if (httpResponse.isPresent()) {
-            int statusCode = httpResponse.get().statusCode();
-            if (statusCode == 401) {
-                this.showInformationDialogAndWait(failedTitle, Localization.lang("Access denied. You are not authorized to access this resource. Please check your credentials and try again. If you believe you should have access, please contact the administrator for assistance.") + "\n\n" + localizedMessage);
-            } else if (statusCode == 403) {
-                this.showInformationDialogAndWait(failedTitle, Localization.lang("Access denied. You do not have permission to access this resource. Please contact the administrator for assistance or try a different action.") + "\n\n" + localizedMessage);
-            } else if (statusCode == 404) {
-                this.showInformationDialogAndWait(failedTitle, Localization.lang("The requested resource could not be found. It seems that the file you are trying to download is not available or has been moved. Please verify the URL and try again. If you believe this is an error, please contact the administrator for further assistance.") + "\n\n" + localizedMessage);
-            } else {
-                this.showErrorDialogAndWait(failedTitle, Localization.lang("Something is wrong on JabRef side. Please check the URL and try again.") + "\n\n" + localizedMessage);
-            }
+            this.showInformationDialogAndWait(failedTitle, Localization.lang(getContentByCode(httpResponse.get().statusCode())) + "\n\n" + localizedMessage);
```

Comment:
No, that does not work.

Localization.lang has to be parameterized with a literal (!) string. Otherwise, our tooling does nt work. Please see https://devdocs.jabref.org/code-howtos/localization.html

Reply:
I found this ToDo in `org.jabref.logic.l10n.LocalizationConsistencyTest`:
```
TODO: Localization.lang(var1 + "test" + var2) not covered
```
In my code is a `var1 + "test" + var2` case, so it failed. maybe I should finish this TODO to pass the unit test?

Reply:
Finishing that ToDo would require several hours of work and design planning. Localization requires full sentences to be translated as in different languages word order and grammar changes d pending on what that var may be. So please just do as @koppor said, fix your strings.

Reply:
> Finishing that ToDo would require several hours of work and design planning. Localization requires full sentences to be translated as in different languages word order and grammar changes d pending on what that var may be. So please just do as @koppor said, fix your strings.

I did as @koppor said, fix my strings before I comment. I just want to make this unit test better by finishing ToDo. And I think it may not require several hours work, such as this simple way:
- assert code in `LocalizationConsistencyTest` now:
```
assertTrue(e.getKey().startsWith("\"") || e.getKey().endsWith("\""), "Illegal localization parameter found. Must include a String with potential concatenation or replacement parameters. Illegal parameter: Localization.lang(" + e.getKey());
```
- a little fix, add `StringUtils.countMatches(e.getKey(), "\"") >= 2` to cover `Localization.lang(var1 + "test" + var2)` case:
```
assertTrue(e.getKey().startsWith("\"") || e.getKey().endsWith("\"") || StringUtils.countMatches(e.getKey(), "\"") >= 2, "Illegal localization parameter found. Must include a String with potential concatenation or replacement parameters. Illegal parameter: Localization.lang(" + e.getKey());
```

Reply:
Discussion follow-up at https://github.com/JabRef/jabref/issues/11784